pkg/codec/www-form-urlencoded/form.go

```diff
@@ -153,5 +155,97 @@ func array(encoder url.Values, root string, refs references.Store, prop *specs.P
 // Unmarshal unmarshals the given www-form-urlencoded io reader into the given reference store.
 // This method is called during runtime to decode a new message and store it inside the given reference store
 func (manager *Manager) Unmarshal(reader io.Reader, refs references.Store) error {
+	if manager.specs == nil {
+		return nil
+	}
+
+	bb, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return err
+	}
+
+	if len(bb) == 0 {
+		return nil
+	}
+
+	values, err := url.ParseQuery(string(bb))
+	if err != nil {
+		return err
+	}
+
+	for key, values := range values {
+		if err := decodeElement(manager.resource, 0, strings.Split(key, "."), values, manager.specs.Nested, refs); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func decodeElement(resource string, pos int, path []string, values []string, schema map[string]*specs.Property, refs references.Store) error {
+	propName := path[pos]
+
+	prop, has := schema[propName]
+	if !has {
+		return errUndefinedProperty(propName)
+	}
+
+	ref := &references.Reference{
+		Path: prop.Path,
+	}
```

Comment:
maybe it is better to check during construction of the codec which paths are allowed to avoid having to look them up inside the property every time. This would simplify the decode method as well.

Reply:
I do not check :thinking: 
I am creating a reference for the given property...